<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Model_name</th>
      <th>Model type</th>
      <th>Average_MCC</th>
      <th>Bootstrap_standard_error_MCC</th>
      <th>Function_Activity</th>
      <th>Function_Binding</th>
      <th>Function_Expression</th>
      <th>Function_OrganismalFitness</th>
      <th>Function_Stability</th>
      <th>Low_MSA_depth</th>
      <th>Medium_MSA_depth</th>
      <th>High_MSA_depth</th>
      <th>Taxa_Human</th>
      <th>Taxa_Other_Eukaryote</th>
      <th>Taxa_Prokaryote</th>
      <th>Taxa_Virus</th>
      <th>Model details</th>
      <th>References</th>
    </tr>
    <tr>
      <th>Model_rank</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>TranceptEVE M</td>
      <td>Hybrid model</td>
      <td>0.387</td>
      <td>0.000</td>
      <td>0.580</td>
      <td>NaN</td>
      <td>0.373</td>
      <td>0.432</td>
      <td>0.164</td>
      <td>0.519</td>
      <td>0.126</td>
      <td>0.251</td>
      <td>0.255</td>
      <td>0.230</td>
      <td>0.146</td>
      <td>0.082</td>
      <td>TranceptEVE Medium model (Tranception Medium & retrieved EVE model)</td>
      <td><a href='https://www.biorxiv.org/content/10.1101/2022.12.07.519495v1?rss=1'>Notin, P., Van Niekerk, L., Kollasch, A., Ritter, D., Gal, Y. & Marks, D.S. &  (2022). TranceptEVE: Combining Family-specific and Family-agnostic Models of Protein Sequences for Improved Fitness Prediction. NeurIPS, LMRL workshop.</a></td>
    </tr>
    <tr>
      <th>2</th>
      <td>TranceptEVE L</td>
      <td>Hybrid model</td>
      <td>0.380</td>
      <td>0.012</td>
      <td>0.544</td>
      <td>NaN</td>
      <td>0.382</td>
      <td>0.381</td>
      <td>0.214</td>
      <td>0.471</td>
      <td>0.144</td>
      <td>0.314</td>
      <td>0.302</td>
      <td>0.257</td>
      <td>0.142</td>
      <td>0.214</td>
      <td>TranceptEVE Large model (Tranception Large & retrieved EVE model)</td>
      <td><a href='https://www.biorxiv.org/content/10.1101/2022.12.07.519495v1?rss=1'>Notin, P., Van Niekerk, L., Kollasch, A., Ritter, D., Gal, Y. & Marks, D.S. &  (2022). TranceptEVE: Combining Family-specific and Family-agnostic Models of Protein Sequences for Improved Fitness Prediction. NeurIPS, LMRL workshop.</a></td>
    </tr>
    <tr>
      <th>3</th>
      <td>Progen2 Base</td>
      <td>Protein language model</td>
      <td>0.375</td>
      <td>0.034</td>
      <td>0.472</td>
      <td>NaN</td>
      <td>0.374</td>
      <td>0.291</td>
      <td>0.365</td>
      <td>0.251</td>
      <td>0.317</td>
      <td>0.416</td>
      <td>0.407</td>
      <td>0.438</td>
      <td>0.236</td>
      <td>0.240</td>
      <td>Progen2 base model (760M params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>4</th>
      <td>Progen2 M</td>
      <td>Protein language model</td>
      <td>0.369</td>
      <td>0.039</td>
      <td>0.423</td>
      <td>NaN</td>
      <td>0.367</td>
      <td>0.296</td>
      <td>0.389</td>
      <td>0.267</td>
      <td>0.358</td>
      <td>0.414</td>
      <td>0.387</td>
      <td>0.450</td>
      <td>0.305</td>
      <td>0.314</td>
      <td>Progen2 medium model (760M params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>5</th>
      <td>Tranception M no retrieval</td>
      <td>Protein language model</td>
      <td>0.368</td>
      <td>0.039</td>
      <td>0.409</td>
      <td>NaN</td>
      <td>0.332</td>
      <td>0.411</td>
      <td>0.319</td>
      <td>0.447</td>
      <td>0.277</td>
      <td>0.366</td>
      <td>0.365</td>
      <td>0.350</td>
      <td>0.217</td>
      <td>0.382</td>
      <td>Tranception Medium model (300M params) without retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>6</th>
      <td>Progen2 L</td>
      <td>Protein language model</td>
      <td>0.359</td>
      <td>0.035</td>
      <td>0.433</td>
      <td>NaN</td>
      <td>0.340</td>
      <td>0.293</td>
      <td>0.371</td>
      <td>0.262</td>
      <td>0.325</td>
      <td>0.414</td>
      <td>0.403</td>
      <td>0.405</td>
      <td>0.287</td>
      <td>0.278</td>
      <td>Progen2 large model (2.7B params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>7</th>
      <td>Tranception L no retrieval</td>
      <td>Protein language model</td>
      <td>0.352</td>
      <td>0.034</td>
      <td>0.361</td>
      <td>NaN</td>
      <td>0.332</td>
      <td>0.371</td>
      <td>0.345</td>
      <td>0.456</td>
      <td>0.275</td>
      <td>0.402</td>
      <td>0.378</td>
      <td>0.378</td>
      <td>0.216</td>
      <td>0.426</td>
      <td>Tranception Large model (700M params) without retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>8</th>
      <td>Hidden Markov Model</td>
      <td>Alignment-based model</td>
      <td>0.351</td>
      <td>0.025</td>
      <td>0.433</td>
      <td>NaN</td>
      <td>0.338</td>
      <td>0.387</td>
      <td>0.245</td>
      <td>0.129</td>
      <td>0.293</td>
      <td>0.255</td>
      <td>0.280</td>
      <td>0.263</td>
      <td>0.246</td>
      <td>0.271</td>
      <td>Profile Hidden Markov model</td>
      <td><a href='http://hmmer.org/'>HMMER: biosequence analysis using profile hidden Markov models</a></td>
    </tr>
    <tr>
      <th>9</th>
      <td>RITA XL</td>
      <td>Protein language model</td>
      <td>0.350</td>
      <td>0.029</td>
      <td>0.401</td>
      <td>NaN</td>
      <td>0.326</td>
      <td>0.319</td>
      <td>0.355</td>
      <td>0.408</td>
      <td>0.331</td>
      <td>0.366</td>
      <td>0.368</td>
      <td>0.378</td>
      <td>0.238</td>
      <td>0.468</td>
      <td>RITA xlarge model (1.2B params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>10</th>
      <td>RITA L</td>
      <td>Protein language model</td>
      <td>0.348</td>
      <td>0.031</td>
      <td>0.375</td>
      <td>NaN</td>
      <td>0.344</td>
      <td>0.325</td>
      <td>0.349</td>
      <td>0.397</td>
      <td>0.325</td>
      <td>0.362</td>
      <td>0.359</td>
      <td>0.369</td>
      <td>0.244</td>
      <td>0.466</td>
      <td>RITA large model (680M params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>11</th>
      <td>Progen2 S</td>
      <td>Protein language model</td>
      <td>0.346</td>
      <td>0.034</td>
      <td>0.451</td>
      <td>NaN</td>
      <td>0.306</td>
      <td>0.294</td>
      <td>0.334</td>
      <td>0.271</td>
      <td>0.300</td>
      <td>0.374</td>
      <td>0.373</td>
      <td>0.363</td>
      <td>0.268</td>
      <td>0.249</td>
      <td>Progen2 small model (150M params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>12</th>
      <td>RITA M</td>
      <td>Protein language model</td>
      <td>0.339</td>
      <td>0.021</td>
      <td>0.388</td>
      <td>NaN</td>
      <td>0.311</td>
      <td>0.327</td>
      <td>0.332</td>
      <td>0.381</td>
      <td>0.273</td>
      <td>0.383</td>
      <td>0.378</td>
      <td>0.334</td>
      <td>0.218</td>
      <td>0.419</td>
      <td>RITA medium model (300M params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>13</th>
      <td>Tranception M</td>
      <td>Hybrid model</td>
      <td>0.334</td>
      <td>0.024</td>
      <td>0.417</td>
      <td>NaN</td>
      <td>0.346</td>
      <td>0.426</td>
      <td>0.145</td>
      <td>0.517</td>
      <td>0.096</td>
      <td>0.231</td>
      <td>0.240</td>
      <td>0.207</td>
      <td>0.097</td>
      <td>0.077</td>
      <td>Tranception Medium model (300M params) with retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>14</th>
      <td>Tranception S no retrieval</td>
      <td>Protein language model</td>
      <td>0.333</td>
      <td>0.033</td>
      <td>0.368</td>
      <td>NaN</td>
      <td>0.312</td>
      <td>0.373</td>
      <td>0.280</td>
      <td>0.414</td>
      <td>0.256</td>
      <td>0.312</td>
      <td>0.280</td>
      <td>0.332</td>
      <td>0.238</td>
      <td>0.324</td>
      <td>Tranception Small model (85M params) without retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>15</th>
      <td>Tranception L</td>
      <td>Hybrid model</td>
      <td>0.329</td>
      <td>0.025</td>
      <td>0.389</td>
      <td>NaN</td>
      <td>0.375</td>
      <td>0.362</td>
      <td>0.190</td>
      <td>0.456</td>
      <td>0.106</td>
      <td>0.291</td>
      <td>0.278</td>
      <td>0.231</td>
      <td>0.096</td>
      <td>0.187</td>
      <td>Tranception Large model (700M params) with retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>16</th>
      <td>Progen2 XL</td>
      <td>Protein language model</td>
      <td>0.326</td>
      <td>0.017</td>
      <td>0.307</td>
      <td>NaN</td>
      <td>0.268</td>
      <td>0.333</td>
      <td>0.396</td>
      <td>0.398</td>
      <td>0.375</td>
      <td>0.393</td>
      <td>0.391</td>
      <td>0.411</td>
      <td>0.338</td>
      <td>0.384</td>
      <td>Progen2 xlarge model (6.4B params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>17</th>
      <td>RITA S</td>
      <td>Protein language model</td>
      <td>0.324</td>
      <td>0.028</td>
      <td>0.383</td>
      <td>NaN</td>
      <td>0.299</td>
      <td>0.327</td>
      <td>0.287</td>
      <td>0.333</td>
      <td>0.264</td>
      <td>0.318</td>
      <td>0.305</td>
      <td>0.309</td>
      <td>0.226</td>
      <td>0.360</td>
      <td>RITA small model (85M params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>18</th>
      <td>TranceptEVE S</td>
      <td>Hybrid model</td>
      <td>0.303</td>
      <td>0.024</td>
      <td>0.427</td>
      <td>NaN</td>
      <td>0.322</td>
      <td>0.391</td>
      <td>0.070</td>
      <td>0.472</td>
      <td>0.057</td>
      <td>0.136</td>
      <td>0.142</td>
      <td>0.137</td>
      <td>0.071</td>
      <td>0.043</td>
      <td>TranceptEVE Small model (Tranception Small & retrieved EVE model)</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>19</th>
      <td>Tranception S</td>
      <td>Hybrid model</td>
      <td>0.291</td>
      <td>0.022</td>
      <td>0.403</td>
      <td>NaN</td>
      <td>0.308</td>
      <td>0.392</td>
      <td>0.061</td>
      <td>0.471</td>
      <td>0.051</td>
      <td>0.125</td>
      <td>0.135</td>
      <td>0.123</td>
      <td>0.062</td>
      <td>0.052</td>
      <td>Tranception Small model (85M params) with retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>20</th>
      <td>Provean</td>
      <td>Alignment-based model</td>
      <td>0.279</td>
      <td>0.024</td>
      <td>0.263</td>
      <td>NaN</td>
      <td>0.298</td>
      <td>0.314</td>
      <td>0.240</td>
      <td>0.354</td>
      <td>0.238</td>
      <td>0.247</td>
      <td>0.205</td>
      <td>0.256</td>
      <td>0.288</td>
      <td>0.301</td>
      <td>Provean model</td>
      <td><a href='https://www.jcvi.org/publications/predicting-functional-effect-amino-acid-substitutions-and-indels'>Choi Y, Sims GE, Murphy S, Miller JR, Chan AP (2012). Predicting the functional effect of amino acid substitutions and indels. PloS one.</a></td>
    </tr>
    <tr>
      <th>21</th>
      <td>Wavenet</td>
      <td>Alignment-based model</td>
      <td>0.237</td>
      <td>0.036</td>
      <td>0.353</td>
      <td>NaN</td>
      <td>0.211</td>
      <td>0.294</td>
      <td>0.089</td>
      <td>0.158</td>
      <td>0.172</td>
      <td>0.072</td>
      <td>0.056</td>
      <td>0.165</td>
      <td>0.106</td>
      <td>0.274</td>
      <td>Wavenet model</td>
      <td><a href='https://www.nature.com/articles/s41467-021-22732-w'>Shin, J., Riesselman, A.J., Kollasch, A.W., McMahon, C., Simon, E., Sander, C., Manglik, A., Kruse, A.C., & Marks, D.S. (2021). Protein design and variant prediction using autoregressive generative models. Nature Communications, 12.</a></td>
    </tr>
    <tr>
      <th>22</th>
      <td>ProtGPT2</td>
      <td>Protein language model</td>
      <td>0.156</td>
      <td>0.028</td>
      <td>0.182</td>
      <td>NaN</td>
      <td>0.196</td>
      <td>0.071</td>
      <td>0.174</td>
      <td>0.285</td>
      <td>0.128</td>
      <td>0.190</td>
      <td>0.229</td>
      <td>0.182</td>
      <td>0.091</td>
      <td>0.046</td>
      <td>ProtGPT2 model</td>
      <td><a href='https://www.nature.com/articles/s41467-022-32007-7'>Ferruz, N., Schmidt, S., & Höcker, B. (2022). ProtGPT2 is a deep unsupervised language model for protein design. Nature Communications, 13.</a></td>
    </tr>
    <tr>
      <th>23</th>
      <td>Unirep</td>
      <td>Protein language model</td>
      <td>0.135</td>
      <td>0.058</td>
      <td>0.098</td>
      <td>NaN</td>
      <td>0.147</td>
      <td>0.030</td>
      <td>0.266</td>
      <td>0.122</td>
      <td>0.210</td>
      <td>0.270</td>
      <td>0.241</td>
      <td>0.288</td>
      <td>0.152</td>
      <td>0.236</td>
      <td>Unirep model</td>
      <td><a href='https://www.nature.com/articles/s41592-019-0598-1'>Alley, E.C., Khimulya, G., Biswas, S., AlQuraishi, M., & Church, G.M. (2019). Unified rational protein engineering with sequence-based deep representation learning. Nature Methods, 1-8.</a></td>
    </tr>
  </tbody>
</table>