<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Model_name</th>
      <th>Model type</th>
      <th>Average_Top_recall</th>
      <th>Bootstrap_standard_error_Top_recall</th>
      <th>Function_Activity</th>
      <th>Function_Binding</th>
      <th>Function_Expression</th>
      <th>Function_OrganismalFitness</th>
      <th>Function_Stability</th>
      <th>Low_MSA_depth</th>
      <th>Medium_MSA_depth</th>
      <th>High_MSA_depth</th>
      <th>Taxa_Human</th>
      <th>Taxa_Other_Eukaryote</th>
      <th>Taxa_Prokaryote</th>
      <th>Taxa_Virus</th>
      <th>Model details</th>
      <th>References</th>
    </tr>
    <tr>
      <th>Model_rank</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>RITA XL</td>
      <td>Protein language model</td>
      <td>0.303</td>
      <td>0.038</td>
      <td>0.260</td>
      <td>NaN</td>
      <td>0.207</td>
      <td>0.301</td>
      <td>0.443</td>
      <td>0.253</td>
      <td>0.385</td>
      <td>0.463</td>
      <td>0.512</td>
      <td>0.346</td>
      <td>0.389</td>
      <td>0.347</td>
      <td>RITA xlarge model (1.2B params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>2</th>
      <td>Progen2 M</td>
      <td>Protein language model</td>
      <td>0.297</td>
      <td>0.036</td>
      <td>0.277</td>
      <td>NaN</td>
      <td>0.223</td>
      <td>0.250</td>
      <td>0.438</td>
      <td>0.174</td>
      <td>0.372</td>
      <td>0.469</td>
      <td>0.490</td>
      <td>0.409</td>
      <td>0.367</td>
      <td>0.205</td>
      <td>Progen2 medium model (760M params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>3</th>
      <td>TranceptEVE L</td>
      <td>Hybrid model</td>
      <td>0.292</td>
      <td>0.012</td>
      <td>0.348</td>
      <td>NaN</td>
      <td>0.176</td>
      <td>0.304</td>
      <td>0.338</td>
      <td>0.294</td>
      <td>0.296</td>
      <td>0.368</td>
      <td>0.356</td>
      <td>0.316</td>
      <td>0.345</td>
      <td>0.256</td>
      <td>TranceptEVE Large model (Tranception Large & retrieved EVE model)</td>
      <td><a href='https://www.biorxiv.org/content/10.1101/2022.12.07.519495v1?rss=1'>Notin, P., Van Niekerk, L., Kollasch, A., Ritter, D., Gal, Y. & Marks, D.S. &  (2022). TranceptEVE: Combining Family-specific and Family-agnostic Models of Protein Sequences for Improved Fitness Prediction. NeurIPS, LMRL workshop.</a></td>
    </tr>
    <tr>
      <th>4</th>
      <td>TranceptEVE M</td>
      <td>Hybrid model</td>
      <td>0.291</td>
      <td>0.000</td>
      <td>0.354</td>
      <td>NaN</td>
      <td>0.199</td>
      <td>0.321</td>
      <td>0.292</td>
      <td>0.249</td>
      <td>0.238</td>
      <td>0.348</td>
      <td>0.315</td>
      <td>0.295</td>
      <td>0.298</td>
      <td>0.195</td>
      <td>TranceptEVE Medium model (Tranception Medium & retrieved EVE model)</td>
      <td><a href='https://www.biorxiv.org/content/10.1101/2022.12.07.519495v1?rss=1'>Notin, P., Van Niekerk, L., Kollasch, A., Ritter, D., Gal, Y. & Marks, D.S. &  (2022). TranceptEVE: Combining Family-specific and Family-agnostic Models of Protein Sequences for Improved Fitness Prediction. NeurIPS, LMRL workshop.</a></td>
    </tr>
    <tr>
      <th>5</th>
      <td>Progen2 XL</td>
      <td>Protein language model</td>
      <td>0.291</td>
      <td>0.033</td>
      <td>0.242</td>
      <td>NaN</td>
      <td>0.169</td>
      <td>0.251</td>
      <td>0.502</td>
      <td>0.235</td>
      <td>0.437</td>
      <td>0.508</td>
      <td>0.512</td>
      <td>0.499</td>
      <td>0.417</td>
      <td>0.255</td>
      <td>Progen2 xlarge model (6.4B params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>6</th>
      <td>Progen2 Base</td>
      <td>Protein language model</td>
      <td>0.290</td>
      <td>0.039</td>
      <td>0.297</td>
      <td>NaN</td>
      <td>0.231</td>
      <td>0.231</td>
      <td>0.402</td>
      <td>0.148</td>
      <td>0.336</td>
      <td>0.439</td>
      <td>0.492</td>
      <td>0.356</td>
      <td>0.302</td>
      <td>0.180</td>
      <td>Progen2 base model (760M params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>7</th>
      <td>Provean</td>
      <td>Alignment-based model</td>
      <td>0.285</td>
      <td>0.037</td>
      <td>0.233</td>
      <td>NaN</td>
      <td>0.162</td>
      <td>0.258</td>
      <td>0.486</td>
      <td>0.199</td>
      <td>0.441</td>
      <td>0.483</td>
      <td>0.446</td>
      <td>0.475</td>
      <td>0.463</td>
      <td>0.358</td>
      <td>Provean model</td>
      <td><a href='https://www.jcvi.org/publications/predicting-functional-effect-amino-acid-substitutions-and-indels'>Choi Y, Sims GE, Murphy S, Miller JR, Chan AP (2012). Predicting the functional effect of amino acid substitutions and indels. PloS one.</a></td>
    </tr>
    <tr>
      <th>8</th>
      <td>RITA L</td>
      <td>Protein language model</td>
      <td>0.275</td>
      <td>0.035</td>
      <td>0.232</td>
      <td>NaN</td>
      <td>0.218</td>
      <td>0.236</td>
      <td>0.413</td>
      <td>0.165</td>
      <td>0.333</td>
      <td>0.455</td>
      <td>0.476</td>
      <td>0.356</td>
      <td>0.309</td>
      <td>0.306</td>
      <td>RITA large model (680M params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>9</th>
      <td>Progen2 L</td>
      <td>Protein language model</td>
      <td>0.271</td>
      <td>0.031</td>
      <td>0.195</td>
      <td>NaN</td>
      <td>0.200</td>
      <td>0.250</td>
      <td>0.439</td>
      <td>0.165</td>
      <td>0.370</td>
      <td>0.466</td>
      <td>0.478</td>
      <td>0.389</td>
      <td>0.409</td>
      <td>0.190</td>
      <td>Progen2 large model (2.7B params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>10</th>
      <td>RITA M</td>
      <td>Protein language model</td>
      <td>0.268</td>
      <td>0.033</td>
      <td>0.210</td>
      <td>NaN</td>
      <td>0.222</td>
      <td>0.251</td>
      <td>0.389</td>
      <td>0.151</td>
      <td>0.308</td>
      <td>0.437</td>
      <td>0.444</td>
      <td>0.308</td>
      <td>0.338</td>
      <td>0.305</td>
      <td>RITA medium model (300M params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>11</th>
      <td>Tranception L no retrieval</td>
      <td>Protein language model</td>
      <td>0.266</td>
      <td>0.034</td>
      <td>0.217</td>
      <td>NaN</td>
      <td>0.184</td>
      <td>0.306</td>
      <td>0.357</td>
      <td>0.269</td>
      <td>0.316</td>
      <td>0.373</td>
      <td>0.388</td>
      <td>0.300</td>
      <td>0.349</td>
      <td>0.287</td>
      <td>Tranception Large model (700M params) without retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>12</th>
      <td>Tranception M no retrieval</td>
      <td>Protein language model</td>
      <td>0.265</td>
      <td>0.031</td>
      <td>0.250</td>
      <td>NaN</td>
      <td>0.205</td>
      <td>0.303</td>
      <td>0.302</td>
      <td>0.207</td>
      <td>0.247</td>
      <td>0.351</td>
      <td>0.329</td>
      <td>0.285</td>
      <td>0.294</td>
      <td>0.217</td>
      <td>Tranception Medium model (300M params) without retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>13</th>
      <td>Progen2 S</td>
      <td>Protein language model</td>
      <td>0.263</td>
      <td>0.038</td>
      <td>0.245</td>
      <td>NaN</td>
      <td>0.238</td>
      <td>0.213</td>
      <td>0.358</td>
      <td>0.129</td>
      <td>0.260</td>
      <td>0.430</td>
      <td>0.412</td>
      <td>0.306</td>
      <td>0.302</td>
      <td>0.240</td>
      <td>Progen2 small model (150M params)</td>
      <td><a href='https://arxiv.org/abs/2206.13517'> Nijkamp, E., Ruffolo, J.A., Weinstein, E.N., Naik, N., & Madani, A. (2022). ProGen2: Exploring the Boundaries of Protein Language Models. ArXiv, abs/2206.13517. </a></td>
    </tr>
    <tr>
      <th>14</th>
      <td>Tranception M</td>
      <td>Hybrid model</td>
      <td>0.259</td>
      <td>0.030</td>
      <td>0.271</td>
      <td>NaN</td>
      <td>0.176</td>
      <td>0.303</td>
      <td>0.287</td>
      <td>0.278</td>
      <td>0.229</td>
      <td>0.335</td>
      <td>0.318</td>
      <td>0.284</td>
      <td>0.258</td>
      <td>0.210</td>
      <td>Tranception Medium model (300M params) with retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>15</th>
      <td>Tranception L</td>
      <td>Hybrid model</td>
      <td>0.257</td>
      <td>0.030</td>
      <td>0.229</td>
      <td>NaN</td>
      <td>0.178</td>
      <td>0.298</td>
      <td>0.323</td>
      <td>0.272</td>
      <td>0.267</td>
      <td>0.360</td>
      <td>0.353</td>
      <td>0.293</td>
      <td>0.300</td>
      <td>0.258</td>
      <td>Tranception Large model (700M params) with retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>16</th>
      <td>RITA S</td>
      <td>Protein language model</td>
      <td>0.252</td>
      <td>0.035</td>
      <td>0.195</td>
      <td>NaN</td>
      <td>0.244</td>
      <td>0.247</td>
      <td>0.323</td>
      <td>0.167</td>
      <td>0.257</td>
      <td>0.371</td>
      <td>0.363</td>
      <td>0.282</td>
      <td>0.287</td>
      <td>0.241</td>
      <td>RITA small model (85M params)</td>
      <td><a href='https://arxiv.org/abs/2205.05789'>Hesslow, D., Zanichelli, N., Notin, P., Poli, I., & Marks, D.S. (2022). RITA: a Study on Scaling Up Generative Protein Sequence Models. ArXiv, abs/2205.05789.</a></td>
    </tr>
    <tr>
      <th>17</th>
      <td>Tranception S no retrieval</td>
      <td>Protein language model</td>
      <td>0.245</td>
      <td>0.037</td>
      <td>0.255</td>
      <td>NaN</td>
      <td>0.247</td>
      <td>0.260</td>
      <td>0.220</td>
      <td>0.184</td>
      <td>0.195</td>
      <td>0.253</td>
      <td>0.260</td>
      <td>0.173</td>
      <td>0.266</td>
      <td>0.146</td>
      <td>Tranception Small model (85M params) without retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>18</th>
      <td>TranceptEVE S</td>
      <td>Hybrid model</td>
      <td>0.235</td>
      <td>0.025</td>
      <td>0.216</td>
      <td>NaN</td>
      <td>0.216</td>
      <td>0.275</td>
      <td>0.234</td>
      <td>0.220</td>
      <td>0.183</td>
      <td>0.284</td>
      <td>0.255</td>
      <td>0.213</td>
      <td>0.259</td>
      <td>0.175</td>
      <td>TranceptEVE Small model (Tranception Small & retrieved EVE model)</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>19</th>
      <td>Tranception S</td>
      <td>Hybrid model</td>
      <td>0.231</td>
      <td>0.027</td>
      <td>0.234</td>
      <td>NaN</td>
      <td>0.230</td>
      <td>0.252</td>
      <td>0.209</td>
      <td>0.203</td>
      <td>0.173</td>
      <td>0.250</td>
      <td>0.232</td>
      <td>0.182</td>
      <td>0.247</td>
      <td>0.154</td>
      <td>Tranception Small model (85M params) with retrieval</td>
      <td><a href='https://proceedings.mlr.press/v162/notin22a.html'>Notin, P., Dias, M., Frazer, J., Marchena-Hurtado, J., Gomez, A.N., Marks, D.S., & Gal, Y. (2022). Tranception: Protein Fitness Prediction with Autoregressive Transformers and Inference-time Retrieval. ICML.</a></td>
    </tr>
    <tr>
      <th>20</th>
      <td>Hidden Markov Model</td>
      <td>Alignment-based model</td>
      <td>0.217</td>
      <td>0.036</td>
      <td>0.229</td>
      <td>NaN</td>
      <td>0.130</td>
      <td>0.219</td>
      <td>0.289</td>
      <td>0.122</td>
      <td>0.296</td>
      <td>0.276</td>
      <td>0.311</td>
      <td>0.262</td>
      <td>0.253</td>
      <td>0.256</td>
      <td>Profile Hidden Markov model</td>
      <td><a href='http://hmmer.org/'>HMMER: biosequence analysis using profile hidden Markov models</a></td>
    </tr>
    <tr>
      <th>21</th>
      <td>Wavenet</td>
      <td>Alignment-based model</td>
      <td>0.188</td>
      <td>0.030</td>
      <td>0.155</td>
      <td>NaN</td>
      <td>0.121</td>
      <td>0.227</td>
      <td>0.250</td>
      <td>0.127</td>
      <td>0.300</td>
      <td>0.200</td>
      <td>0.133</td>
      <td>0.322</td>
      <td>0.279</td>
      <td>0.326</td>
      <td>Wavenet model</td>
      <td><a href='https://www.nature.com/articles/s41467-021-22732-w'>Shin, J., Riesselman, A.J., Kollasch, A.W., McMahon, C., Simon, E., Sander, C., Manglik, A., Kruse, A.C., & Marks, D.S. (2021). Protein design and variant prediction using autoregressive generative models. Nature Communications, 12.</a></td>
    </tr>
    <tr>
      <th>22</th>
      <td>ProtGPT2</td>
      <td>Protein language model</td>
      <td>0.154</td>
      <td>0.030</td>
      <td>0.138</td>
      <td>NaN</td>
      <td>0.134</td>
      <td>0.110</td>
      <td>0.237</td>
      <td>0.110</td>
      <td>0.164</td>
      <td>0.280</td>
      <td>0.327</td>
      <td>0.185</td>
      <td>0.130</td>
      <td>0.113</td>
      <td>ProtGPT2 model</td>
      <td><a href='https://www.nature.com/articles/s41467-022-32007-7'>Ferruz, N., Schmidt, S., & Höcker, B. (2022). ProtGPT2 is a deep unsupervised language model for protein design. Nature Communications, 13.</a></td>
    </tr>
    <tr>
      <th>23</th>
      <td>Unirep</td>
      <td>Protein language model</td>
      <td>0.153</td>
      <td>0.027</td>
      <td>0.135</td>
      <td>NaN</td>
      <td>0.166</td>
      <td>0.115</td>
      <td>0.197</td>
      <td>0.126</td>
      <td>0.142</td>
      <td>0.234</td>
      <td>0.212</td>
      <td>0.170</td>
      <td>0.139</td>
      <td>0.257</td>
      <td>Unirep model</td>
      <td><a href='https://www.nature.com/articles/s41592-019-0598-1'>Alley, E.C., Khimulya, G., Biswas, S., AlQuraishi, M., & Church, G.M. (2019). Unified rational protein engineering with sequence-based deep representation learning. Nature Methods, 1-8.</a></td>
    </tr>
  </tbody>
</table>